src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (a) => {
+    if (typeof a === 'undefined') {
+      const returnedValue = sum;
+
+      sum = 0;
+
+      return returnedValue;
+    }
+
+    sum += a;
+
+    return adder;
```

Comment:
The parameter 'a' should be given a more descriptive name, such as 'value' or 'numberToAdd', to improve code readability.